test/unit/multiplication/test_multiplication_general.cpp

```diff
@@ -27,18 +30,177 @@
+  const auto fullValuesC = mix_values(config.sub_c(), subValuesC, [](auto) { return T(-99); });
+
+  Matrix<const T, Device::CPU> mat_ah = setMatrix(fullValuesA, config.full_a(), {config.mb, config.kb});
+  Matrix<const T, Device::CPU> mat_bh = setMatrix(fullValuesB, config.full_b(), {config.kb, config.nb});
+  Matrix<T, Device::CPU> mat_ch = setMatrix(fullValuesC, config.full_c(), {config.mb, config.nb});
+
+  {
+    MatrixMirror<const T, D, Device::CPU> mat_a(mat_ah);
+    MatrixMirror<const T, D, Device::CPU> mat_b(mat_bh);
+    MatrixMirror<T, D, Device::CPU> mat_c(mat_ch);
+
+    MatrixRef<const T, D> mat_sub_a(mat_a.get(), config.sub_a());
+    MatrixRef<const T, D> mat_sub_b(mat_b.get(), config.sub_b());
+    MatrixRef<T, D> mat_sub_c(mat_c.get(), config.sub_c());
+
+    // Note: currently it is implemented just the NoTrans/NoTrans case
+    ASSERT_EQ(config.opA, blas::Op::NoTrans);
+    ASSERT_EQ(config.opB, blas::Op::NoTrans);
+    multiplication::internal::General<B, D, T>::callNN(alpha, mat_sub_a, mat_sub_b, beta, mat_sub_c);
+  }
+
+  const auto fullValuesResult = mix_values(config.sub_c(), subValuesResult, fullValuesC);
+  CHECK_MATRIX_NEAR(fullValuesResult, mat_ch, 2 * (mat_ah.size().cols() + 1) * TypeUtilities<T>::error,
+                    2 * (mat_ah.size().cols() + 1) * TypeUtilities<T>::error);
+}
+
+std::vector<GemmConfig> gemm_configs = {
+    // empty matrices
+    {blas::Op::NoTrans, blas::Op::NoTrans, 0, 0, 7, 3, 6, 2},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 26, 0, 7, 3, 6, 2},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 0, 13, 7, 3, 6, 2},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 26, 13, 0, 3, 6, 2},
+
+    // full
+    {blas::Op::NoTrans, blas::Op::NoTrans, 3, 3, 3, 3, 3, 3},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 8, 8, 11, 10, 9, 13},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 3, 2, 4, 1, 1, 1},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 6, 9, 8, 2, 3, 4},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 21, 21, 21, 3, 4, 5},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 12, 20, 11, 3, 4, 5},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 8, 8, 11, 3, 3, 5},
+};
+
+std::vector<GemmConfig> sub_gemm_configs = {
+    // empty matrices
+    {blas::Op::NoTrans, blas::Op::NoTrans, 0, 0, 7, 3, 6, 2, {{1, 2}}, {{2, 3}}, {{3, 4}}},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 26, 0, 7, 3, 6, 2, {{1, 2}}, {{2, 3}}, {{3, 4}}},
+    {blas::Op::NoTrans, blas::Op::NoTrans, 0, 13, 7, 3, 6, 2, {{1, 2}}, {{2, 3}}, {{3, 4}}},
+    // k = 0
+    {blas::Op::NoTrans, blas::Op::NoTrans, 26, 13, 0, 3, 6, 2, {{1, 2}}, {{2, 3}}, {{3, 4}}},
+    // single-tile
+    {blas::Op::NoTrans, blas::Op::NoTrans, 8, 8, 11, 10, 9, 13, {{2, 1}}, {{1, 1}}, {{0, 0}}},
+    // multi-tile
+    {blas::Op::NoTrans, blas::Op::NoTrans, 12, 20, 11, 3, 4, 5, {{7, 1}}, {{11, 10}}, {{4, 2}}},
+};
+
+TYPED_TEST(GeneralMultiplicationTestMC, CorrectnessLocalWithMatrixRef) {
```

Comment:
Probably better dropping `MatrixRef`?

```suggestion
TYPED_TEST(GeneralMultiplicationTestMC, CorrectnessLocal) {
```